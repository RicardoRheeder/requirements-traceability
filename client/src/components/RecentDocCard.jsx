import React, { useEffect } from 'react'
import { useDispatch, useSelector, dispatch } from 'react-redux'
import { DocumentPanel } from './DocumentPanel'
import { useAuth0 } from '@auth0/auth0-react'
import { fetchUserRecentDocsAsync } from '../redux/stores/user/actions'

export default function RecentDocCard({}) {
  const dispatch = useDispatch()
  const recent_docs = useSelector((state) => state.user.recent_docs)
  const renderRecentDocs = (recent_docs) => {
    if (recent_docs != null) {
      return recent_docs.map((recent_doc, i) => {
        return <DocumentPanel document={recent_doc} key={i} />
      })
    }
    return (
      <div className="left-container-placeholder">
        <h2>No existing documents.</h2>
        <div>Click the 'Add' button below to get started.</div>
      </div>
    )
  }
  return <div>{renderRecentDocs(recent_docs)}</div>
}
