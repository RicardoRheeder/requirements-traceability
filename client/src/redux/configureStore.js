import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

// restricting extension in production using process.env.NODE_ENV which is set 'production'
import { composeWithDevTools } from 'redux-devtools-extension/logOnlyInProduction';

import store from './stores';

const enhancer = (process.env.NODE_ENV == 'development')
               ? composeWithDevTools(applyMiddleware(thunk))
               : applyMiddleware(thunk);

export default () => createStore(store, enhancer);
