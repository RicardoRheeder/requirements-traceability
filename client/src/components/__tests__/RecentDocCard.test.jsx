import React from "react"
import Enzyme, { shallow, mount} from "enzyme"
import RecentDocCard from "../RecentDocCard"
import Adapter from "enzyme-adapter-react-16"
import { Provider } from "react-redux";
import configureStore from "redux-mock-store"


const mockStore = configureStore([])
Enzyme.configure({adapter: new Adapter() });

describe("RecentDocCard", () => {
    let store;
    beforeEach(() => {
        store = mockStore({
            document: {
                documents: [{title: 'testdoc', versions: [], tree: null}],
                versions: []
            },
            common: {
                selectedDocumentPanelObject: {title: 'testdoc'}
            },
            user: {
               recent_docs: []
            }
            
        });
    });
    test("renders", () => {
        const wrapper = shallow(<Provider store = {store}><RecentDocCard/></Provider>);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
    test("renders with children", () => {
        const wrapper = mount(<Provider store = {store}><RecentDocCard/></Provider>);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
});