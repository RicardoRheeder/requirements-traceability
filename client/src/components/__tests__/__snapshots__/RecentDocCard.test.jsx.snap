// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentDocCard renders 1`] = `ShallowWrapper {}`;

exports[`RecentDocCard renders with children 1`] = `ReactWrapper {}`;
